homeassistant/components/light/insteon_local.py

```diff
@@ -0,0 +1,125 @@
+"""
+Support for Insteon local lights.
+
+For more details about this platform, please refer to the documentation at
+
+--
+Example platform config
+--
+
+insteon_local:
+  host: YOUR HUB IP
+  username: YOUR HUB USERNAME
+  password: YOUR HUB PASSWORD
+
+--
+Example platform config
+--
+
+light:
+  - platform: insteon_local
+    lights:
+      dining_room:
+        device_id: 30DA8A
+        name: Dining Room
+      living_room:
+        device_id: 30D927
+        name: Living Room
+
+"""
+
+from time import sleep
+from datetime import timedelta
+from homeassistant.components.light import (ATTR_BRIGHTNESS,
+                                            SUPPORT_BRIGHTNESS, Light)
+import homeassistant.util as util
+
+
+DEPENDENCIES = ['insteon_local']
+
+SUPPORT_INSTEON_LOCAL = SUPPORT_BRIGHTNESS
+
+MIN_TIME_BETWEEN_SCANS = timedelta(seconds=5)
+MIN_TIME_BETWEEN_FORCED_SCANS = timedelta(milliseconds=100)
+
+DOMAIN = "light"
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the Insteon local light platform."""
+    insteonhub = hass.data['insteon_local']
+    devs = []
+    if len(config) > 0:
+        items = config['lights'].items()
+
+        # todo: use getLinked instead
+        for light in items:
+            device = insteonhub.dimmer(light[1]['device_id'])
+            device.beep()
+            devs.append(InsteonLocalDimmerDevice(device, light[1]['name']))
+        add_devices(devs)
+
+
+class InsteonLocalDimmerDevice(Light):
+    """An abstract Class for an Insteon node."""
+
+    def __init__(self, node, name):
+        """Initialize the device."""
+        self.node = node
+        self.node.deviceName = name
+        self._value = 0
+
+    @property
+    def name(self):
+        """Return the the name of the node."""
+        return self.node.deviceName
+
+    @property
+    def unique_id(self):
+        """Return the ID of this insteon node."""
+        return self.node.deviceId
+
+    @property
+    def brightness(self):
+        """Return the brightness of this light between 0..255."""
+        return self._value
+
+    @util.Throttle(MIN_TIME_BETWEEN_SCANS, MIN_TIME_BETWEEN_FORCED_SCANS)
+    def update(self):
+        """Update state of the sensor."""
+        devid = self.node.deviceId.upper()
```

Comment:
This is protocol implementation and should be integrated into the insteon_local lib and removed from Home Assistant. Home Assistant should only call `self.node.update()` or something similar.